shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/data/local/datasource/TokenLocalDataSource.kt

```diff
@@ -0,0 +1,37 @@
+package vn.luongvo.kmm.survey.data.local.datasource
+
+import com.russhwolf.settings.*
+import vn.luongvo.kmm.survey.domain.model.Token
+
+private const val TOKEN_TYPE_KEY = "tokenType"
+private const val ACCESS_TOKEN_KEY = "accessToken"
+private const val REFRESH_TOKEN_KEY = "refreshToken"
+
+interface TokenLocalDataSource {
+
+    fun saveToken(token: Token)
+
+    val tokenType: String
+
+    val accessToken: String
+
+    val refreshToken: String
+}
+
+class TokenLocalDataSourceImpl(private val settings: Settings) : TokenLocalDataSource {
+
+    override fun saveToken(token: Token) = with(token) {
+        settings[TOKEN_TYPE_KEY] = tokenType
```

Comment:
You can use https://github.com/russhwolf/multiplatform-settings#serialization-module to save objects instead of manually save them, if that's what you're looking for.

Reply:
@blyscuit I saw that support, however with these simple 3 key/value, I don't think it's good to apply because of lower performance for sure 🤓